### PR TITLE
Sync Files: Bring changes from WPCOM to class.jetpack-sync-defaults.php

### PR DESCRIPTION
**Pull request by lezama**
**DON'T MERGE FOR 5.1**

#### Changes proposed in this Pull Request:

* Sync WPCOM changes to class.jetpack-sync-defaults.php

#### Testing instructions:

* Test all of the things sync

### CODE REVIEW

File: sync/class.jetpack-sync-defaults.php [hunk 1]
```diff
@@ -237,14 +296,19 @@ public static function get_callable_whitelist() {
...
-		'meta_value'
+		'meta_value',
 	);
 
 	static $default_option_checksum_columns = array(
 		'option_name',
 		'option_value',
 	);
 
+	// returns escapted SQL that can be injected into a WHERE clause
```
Comment (roccotripaldi) [inline sync/class.jetpack-sync-defaults.php]: ^ escaped

File: sync/class.jetpack-sync-defaults.php [hunk 2]
```diff
@@ -128,6 +129,49 @@ public static function get_options_whitelist() {
 		return apply_filters( 'jetpack_sync_options_whitelist', $options_whitelist );
 	}
 
+	/**
+	 * These are options that don't need to go in the regular wp_options table.
+	 * The reason we're putting them here is that all wp_options are pre-loaded to memory and we don't want to abuse that.
+	 *
```
Comment (roccotripaldi) [inline sync/class.jetpack-sync-defaults.php]: ❤️

**Comment by zinigor** [comment]
It looks like the PR broke Travis. Travis wants another test to cover one of the options.
If you wanted to hold this PR for the 5.2 cycle, you can safely merge it now, if you feel it's ready.

**Comment by enejb** [comment]
I don't think we should merge this since in Jetpack we have conditions that check that Woo is also enabled and only load the module when that is the case.

**Comment by dereksmart** [comment]
This PR is quite old and @enejb doesn't think we should merge.  @lezama please re-open if you still want this in.

**Comment by enejb** [comment]
I think it might make more sense to separate out the woo part from`class.jetpack-sync-defaults.php`. So that it doesn't have to synced like this.

**Comment by lezama** [comment]
> This PR is quite old and @enejb doesn't think we should merge. @lezama please re-open if you still want this in.

ok :)

**Comment by lezama** [comment]
> I think it might make more sense to separate out the woo part from class.jetpack-sync-defaults.php. So that it doesn't have to synced like this.

Agreed, but we could do that on another PR, we should try to keep both sides in sync.

**Comment by lezama** [comment]
@dereksmart probably this is not up to date anymore, I think we could close this if having this file in sync with WPCOM is reflected on another issue.

**Comment by dereksmart** [comment]
Way out of date :) we'll be handling in other PRs

**Pull request closed without merging**